### PR TITLE
MESG token integration

### PR DESCRIPTION
**Pull request by antho1404**

### CODE REVIEW

File: config/key.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Key is the structure that stores all informations related to the key system
+type Key struct {
+	Address string
+}
+
+// GetKey returns the core's key
+func (c *Config) GetKey() (*Key, error) {
```
Comment (krhubert) [inline config/key.go]: This one could be loaded when whole config is load (at once)

File: interface/grpc/core/create_workflow.go [hunk 2]
```diff
@@ -8,6 +8,9 @@ import (
 
 // CreateWorkflow creates and runs a new workflow.
 func (s *Server) CreateWorkflow(ctx context.Context, request *coreapi.CreateWorkflowRequest) (*coreapi.CreateWorkflowReply, error) {
+	if err := s.requireStake(); err != nil {
```
Comment (krhubert) [inline interface/grpc/core/create_workflow.go]: for requireStake grpc-middelware would be great

File: interface/grpc/core/require_stake.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package core
+
+import "github.com/mesg-foundation/core/config"
+
+// ListenEvent listens events matches with eventFilter on serviceID.
+func (s *Server) requireStake() error {
+	c, err := config.Global()
```
Comment (krhubert) [inline interface/grpc/core/require_stake.go]: Do not use config.Global, pass it as argument

File: interface/grpc/core/require_stake.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+	if err != nil {
+		return err
+	}
+	k, err := c.GetKey()
```
Comment (krhubert) [inline interface/grpc/core/require_stake.go]: Key will be loaded from disk every time someone call api.

**Comment by NicolasMahe** [comment]
Let's keep it open for now. I will close it when i don't need it.

**Comment by NicolasMahe** [comment]
Closing this PR as it's not relevant anymore

**Pull request closed without merging**